python/lsst/ts/electrometer/controller.py

```diff
@@ -333,47 +360,117 @@ async def start_scan_dt(self, scan_duration):
 
-    async def write_fits_file(self, intensity, times, temperature, unit, voltage):
+    def make_primary_header(self):
+        """Make primary header for fits file that follows Rubin Obs. format."""
+        primary_hdu = fits.PrimaryHDU()
+        primary_hdu.header["FORMAT_V"] = ("1", "Header format version")
+        primary_hdu.header["OBSERVAT"] = "Vera C. Rubin Observatory"
+        primary_hdu.header["INSTRUME"] = (
+            f"Electrometer_index_{self.csc.salinfo.index}",
+            "Type of Instrument",
+        )
+        primary_hdu.header["MODEL"] = (self.model_id, "Model of instrument")
+        primary_hdu.header["LOCATN"] = (self.location, "Location of Instrument")
+        primary_hdu.header["ORIGIN"] = (
+            self.csc.salinfo.name,
+            "Name of the program that produced this data.",
+        )
+        primary_hdu.header["DATE-BEG"] = (
+            self.manual_start_time,
+            "When start scan command sent to CSC (TAI)",
+        )
+        primary_hdu.header["DATE-END"] = (
+            self.manual_end_time,
+            "When stop scan command sent to CSC (TAI)",
+        )
+        primary_hdu.header["TIMESYS"] = ("TAI", "Format of timestamps")
+        primary_hdu.header["SCANTIME"] = (self.scan_duration, "Duration of scan [s]")
+        primary_hdu.header["SAMPTIME"] = (
+            self.integration_time,
+            "Duration of each sample [s]",
+        )
+        primary_hdu.header["FILTMED"] = (
+            self.median_filter_active,
+            "Median Filter Active",
+        )
+        primary_hdu.header["FILTAVG"] = (
+            self.avg_filter_active,
+            "Average Filter Active",
+        )
+        primary_hdu.header["IMGTYPE"] = (
+            self.mode,
+            "Options are charge, voltage, current",
+        )
+        primary_hdu.header["SENSBRND"] = (self.sensor_brand, "Sensor brand")
+        primary_hdu.header["SENSMODL"] = (self.sensor_model, "Sensor model")
+        primary_hdu.header["SERIAL"] = (self.sensor_serial, "Sensor serial number")
+        primary_hdu.header["TEMP"] = (
+            self.temperature,
+            "Measurement from probe if attached and declared (Celcius)",
+        )
+        primary_hdu.header["VSOURCE"] = (
+            self.vsource,
+            "Voltage input if active and attached",
+        )
+        return primary_hdu
+
+    async def write_fits_file(self, signal, times, temperature, unit, voltage):
         """Write fits file of the intensity, time, and temperature values.
 
         Parameters
         ----------
-        intensity : `list`
-            The intensity values
-        times : `list`
-            The time of the intensity value
-        temperature : `list`
-            The temperature of the intensity value
-        unit : `list`
-            The unit of the intensity value.
+        signal : `list` of `float`
+            The amount of photons in a given reading, unit depends on mode of
```

Comment:
Please add here an explanation for which modes there are, what data the modes produce and what unit they have.